### PR TITLE
[SPEC] NonMaxSuppression

### PR DESCRIPTION
**Pull request by tomdol**

### CODE REVIEW

File: src/ngraph/frontend/onnx_import/op/non_max_suppression.cpp [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                    const std::shared_ptr<ngraph::Node> scores = ng_inputs.at(1);
+
+                    std::shared_ptr<ngraph::Node> max_output_boxes_per_class =
+                        ngraph::op::Constant::create(element::i64, Shape{}, {0});
+                    if (ng_inputs.size() > 2)
+                    {
+                        max_output_boxes_per_class = ng_inputs.at(2);
+                    }
+
+                    std::shared_ptr<ngraph::Node> iou_threshold =
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/op/non_max_suppression.cpp]: We don't need to create these constants by default.

File: src/ngraph/frontend/onnx_import/op/non_max_suppression.cpp [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                NodeVector non_max_suppression(const Node& node)
+                {
+                    // TODO: this op will not be tested until at least
+                    //       a reference implementation is added
+
+                    const auto ng_inputs = node.get_ng_inputs();
+                    const std::shared_ptr<ngraph::Node> boxes = ng_inputs.at(0);
+                    const std::shared_ptr<ngraph::Node> scores = ng_inputs.at(1);
+
+                    std::shared_ptr<ngraph::Node> max_output_boxes_per_class =
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/op/non_max_suppression.cpp]: We don't need to create these constants by default.
You can create them in the `else` clause.

File: src/ngraph/frontend/onnx_import/op/non_max_suppression.cpp [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                    }
+
+                    std::shared_ptr<ngraph::Node> iou_threshold =
+                        ngraph::op::Constant::create(element::f32, Shape{}, {.0f});
+                    if (ng_inputs.size() > 3)
+                    {
+                        iou_threshold = ng_inputs.at(3);
+                    }
+
+                    std::shared_ptr<ngraph::Node> score_threshold =
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/op/non_max_suppression.cpp]: We don't need to create these constants by default.

File: src/ngraph/op/non_max_suppression.hpp [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+                /// \brief Constructs a NonMaxSuppression operation with default values for the last
+                ///        3 inputs
+                ///
+                /// \param boxes Output that produces a tensor with box coordinates.
+                /// \param scores Output that produces ta tensor
+                /// \param box_encoding Auto broadcast specification
+                /// \param sort_result_descending Auto broadcast specification
+                NonMaxSuppression(const Output<Node>& boxes,
+                                  const Output<Node>& scores,
+                                  const BoxEncodingType box_encoding = BoxEncodingType::CORNER,
```
Comment (ilyachur) [inline src/ngraph/op/non_max_suppression.hpp]: In the spec we have 3 required inputs: boxes, scores and score_threshold.

Comment (tomdol) [reply src/ngraph/op/non_max_suppression.hpp]: The current version of the spec excplicitly mentions that `max_output_boxes_per_class` is optional. Can you double check?

Comment (ilyachur) [reply src/ngraph/op/non_max_suppression.hpp]: Yes, sorry, my fault.  score_threshold - Optional with default value 0

**Pull request merged**